api/src/prescription/learner-management/infrastructure/repositories/organization-learner-repository.js

```diff
@@ -78,10 +79,60 @@ const disableCommonOrganizationLearnersFromOrganizationId = function ({ organiza
     .whereNotIn('id', organizationLearnerIds);
 };
 
+const findAllCommonLearnersFromOrganizationIdAndUnicityKeys = async function ({ organizationId, unicityKeys }) {
+  const knex = ApplicationTransaction.getConnection();
+
+  const query = knex('view-active-organization-learners')
+    .select(['firstName', 'id', 'lastName', 'userId', 'organizationId', 'attributes'])
+    .where({ organizationId });
+
+  if (unicityKeys) {
+    const attributes = [];
+    const firstNames = [];
+    const lastNames = [];
+
+    unicityKeys.forEach((unicityKey) => {
+      if (unicityKey.firstName) lastNames.push(unicityKey.firstName);
+      if (unicityKey.lastName) lastNames.push(unicityKey.lastName);
+
+      const unicityAttributes = unicityKey;
+      delete unicityAttributes['firsName'];
+      delete unicityAttributes['lastName'];
+
+      attributes.push(unicityAttributes);
+    });
+
+    query.andWhere(function (main_querybuilder) {
+      if (attributes.length > 0) {
+        attributes.forEach((attribute, index) => {
+          if (index === 0) {
+            main_querybuilder.whereJsonSupersetOf('attributes', attribute);
+          } else {
+            main_querybuilder.orWhereJsonSupersetOf('attributes', attribute);
+          }
+        });
+      }
+    });
+
+    if (firstNames.length > 0) {
+      query.whereIn('firstName', firstNames);
+    }
+
+    if (lastNames.length > 0) {
+      query.whereIn('lastName', lastNames);
+    }
```

Comment:
j'(ai un doute sur l'écriture de cette requete . 

ça devrait plutôt être and `(json firsname lastname) or (json firstname lastname) or ....` .

j'ai un doute sur la bonne seggregation des données avec un `(json or json or json) and (lastnames and firstnames)`

Reply:
cette suggestion n'a plus d'intêret puisque cette restriction de données n'est pas nécessaire.